### PR TITLE
feat: switch to emotion css in js

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [x] - Latest `master` branch merged

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

MIgrate away from two less maintained packages to equivalents which better support the incoming server-sider rendered flow we want to offer for [logged out pages](https://github.com/ONEARMY/community-platform/issues/1418)

1. [`styled-components`](https://snyk.io/advisor/npm-package/styled-components) to [`emotion`](https://snyk.io/advisor/npm-package/@emotion/react)
2. [`rebass`](https://snyk.io/advisor/npm-package/rebass) to [`theme-ui`](https://snyk.io/advisor/npm-package/theme-ui)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -33,7 +32,7 @@ This project and everyone participating in it is governed by the [Code of Conduc
 
 ### Frontend
 
-The platform is built in **React** and [Typescript](https://www.typescriptlang.org/docs/handbook/basic-types.html) for typing and consistency, [styled-components](https://www.styled-components.com/) for reusability and component based styling.
+The platform is built in **React** and [Typescript](https://www.typescriptlang.org/docs/handbook/basic-types.html) for typing and consistency, [Emotion](https://emotion.sh/) for reusability and component based styling.
```
Comment (chrismclarke) [inline CONTRIBUTING.md:35]: Wow, even updating docs 🤩

File: .circleci/config.yml [hunk 2]
```diff
@@ -137,9 +137,9 @@ commands:
     # parameters:
     steps:
       - checkout
-      - *restore_yarn_cache
+      # - *restore_yarn_cache
```
Comment (chrismclarke) [inline .circleci/config.yml:140]: I'm guessing this is related to some of the caching issues you had. I'm assuming this is the best fix we have for now which is fine to help get this merged in. If you want any help trying to figure out after just let me know

File: package.json [hunk 3]
```diff
@@ -14,7 +14,8 @@
   "private": true,
   "main": "lib/index.js",
   "scripts": {
-    "start": "yarn workspace oa-components dev & cross-env FAST_REFRESH=false yarn react-scripts start",
+    "start": "yarn workspace oa-components dev & yarn start:platform",
```
Comment (chrismclarke) [inline package.json:17]: sweeet, nice to have live-reload work with components also

File: package.json [hunk 4]
```diff
@@ -119,11 +122,10 @@
     "react-table": "^6.10.3",
     "react-tooltip": "^4.2.20",
     "react-virtualized": "^9.22.3",
-    "rebass": "4.0.7",
     "rxjs": "^6.6.3",
     "sass": "^1.45.2",
-    "styled-components": "^5.3.0",
     "styled-system": "^5.1.5",
```
Comment (chrismclarke) [inline package.json:127]: Not sure if we still need the `styled-system` dep after removing rebass and styled-components, but can always just do a depcheck after merge (expect not the only one)

File: packages/components/.storybook/main.js [hunk 5]
```diff
@@ -5,10 +5,13 @@ const paths = {
 
 module.exports = {
   stories: [
-    '../../../src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
-    '../../components/src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
+    // '../../../src/**/*.stories.@(js|jsx|ts|tsx|mdx)',
```
Comment (chrismclarke) [inline packages/components/.storybook/main.js:8]: This feels like the world's hardest game of spot the difference... To me the commented out looks identical to the left in so maybe it should go? But I might just be blind. Either way, non-blocking

File: src/components/Comment/CommentTextArea.tsx [hunk 6]
```diff
@@ -33,6 +33,7 @@ const TextAreaStyled = styled.textarea`
 const BoxStyled = styled(Box)`
   position: relative;
   border-radius: 5px;
+  display: block;
```
Comment (chrismclarke) [inline src/components/Comment/CommentTextArea.tsx:36]: surprised a box doesn't block display by default, definitely a pain to have to update all the styled components but thanks for taking the time to do so

**Review by chrismclarke — APPROVED** [review]
Thanks for this @thisislawatts , I know the task grew far away from what was likely expected (I definitely felt a lot of review empathy seeing all the inline properties having to be extracted into sx tags on top of de-nesting some of the deep flex boxes on top of all the more general replacements. 

Thanks for taking so much time to do all this, your find-replace skills are second to none! I'd be amazed if there weren't a few minor inconsistencies, but really I think this is already more than we could have expected given the lack of any automated visual regression testing system and the scale of changes required. Here's hoping we don't have to do it again for some time to come.

File: .circleci/config.yml (hunk 2 above)
Comment (thisislawatts) [reply .circleci/config.yml]: Reverted

File: packages/components/package.json [hunk 7]
```diff
@@ -9,29 +9,33 @@
     "start": "start-storybook -p 6006",
     "format": "prettier --write '**/*.{tsx,mdx}'",
     "build:sb": "yarn build && build-storybook",
-    "build": "tsc",
+    "build": "tsc --build --verbose",
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply packages/components/package.json]: Reverted

**Pull request merged**